File: medicfy/lib/src/models/duration.dart
class Duration {
  int durationId;
  String initialDate;
  String finalDate;
  bool haveFinalDate;
  Duration(
      {this.initialDate, this.finalDate, this.durationId, this.haveFinalDate});
  Map<String, dynamic> toMap() {
    Map<String, dynamic> map = new Map();
    map['durationId'] = durationId;
    map['initialDate'] = initialDate;
    map['finalDate'] = finalDate;
    map['haveFinalDate'] = haveFinalDate;

    return map;
  }
}

File: BackEndMedicFy/Models/Duration.cs
using System.ComponentModel.DataAnnotations;
namespace BackEndMedicFy.Models
{
    public class Duration
    {
        [Key]
        public int DurationId { get; set; }
        public string InitialDate { get; set; }
        public bool haveFinalDate { get; set; }
        public string FinalDate { get; set; }
    }
}